src/main/scala/mesosphere/marathon/core/task/tracker/impl/steps/ContinueOnErrorStep.scala

```diff
@@ -18,13 +18,21 @@ class ContinueOnErrorStep(wrapped: TaskStatusUpdateStep) extends TaskStatusUpdat
   override def name: String = s"continueOnError(${wrapped.name})"
 
   override def processUpdate(
-    timestamp: Timestamp, appId: PathId, maybeTask: Option[MarathonTask], mesosStatus: TaskStatus): Future[_] = {
+    timestamp: Timestamp, appId: PathId, task: MarathonTask, mesosStatus: TaskStatus): Future[_] = {
 
     import scala.concurrent.ExecutionContext.Implicits.global
-    wrapped.processUpdate(timestamp, appId, maybeTask, mesosStatus).recover {
-      case NonFatal(e) =>
-        log.error("while executing step {} for [{}], continue with other steps",
-          wrapped.name, mesosStatus.getTaskId.getValue, e)
+    val maybeProcessed: Option[Future[_]] = Option(wrapped.processUpdate(timestamp, appId, task, mesosStatus))
```

Comment:
We fixed the test. But if we have a faulty test or faulty production code we will get better error feedback.